x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -67,10 +77,43 @@ func (k Keeper) HandleConsumerMisbehaviour(ctx sdk.Context, misbehaviour ibctmty
 	return nil
 }
 
+// CheckMisbehaviourAndUpdateState checks for client misbehaviour and freeze the client if it's still active.
+// Note that this method changes the original CheckMisbehaviourAndUpdateState method in ibc-go ibc-go/modules/core/02-client/keeper/client.go
+func (k Keeper) CheckMisbehaviourAndUpdateState(ctx sdk.Context, misbehaviour exported.Misbehaviour) error {
+	if err := misbehaviour.ValidateBasic(); err != nil {
+		return err
+	}
+
+	// get client state
+	clientState, found := k.clientKeeper.GetClientState(ctx, misbehaviour.GetClientID())
+	if !found {
+		return sdkerrors.Wrapf(ibcclienttypes.ErrClientNotFound, "cannot check misbehaviour for client with ID %s", misbehaviour.GetClientID())
+	}
+
+	clientStore := k.clientKeeper.ClientStore(ctx, misbehaviour.GetClientID())
+
+	// check the IBC misbehaviour
+	_, err := clientState.CheckMisbehaviourAndUpdateState(ctx, k.cdc, clientStore, misbehaviour)
+	if err != nil {
+		return err
+	}
+
+	// Check that the client's status isn't Expired
+	consState, err := ibctmtypes.GetConsensusState(clientStore, k.cdc, clientState.GetLatestHeight())
+	if err != nil {
+		return err
+	}
+
+	if !clientState.(*ibctmtypes.ClientState).IsExpired(consState.Timestamp, ctx.BlockTime()) {
+		k.Logger(ctx).Info("client frozen due to a consumer chain misbehaviour", "client-id", misbehaviour.GetClientID())
```

Comment:
```suggestion
		k.Logger(ctx).Info("client is expired", "client-id", misbehaviour.GetClientID())
```

Reply:
`CheckMisbehaviourAndUpdateState` above already checks that the consensus states at trusted heights are within trusting period so this check seems redundant. If we want to keep it we should move it above `CheckMisbehaviourAndUpdateState`.